src/main/java/com/github/firmwehr/gentle/firm/construction/EntityHelper.java

```diff
@@ -0,0 +1,83 @@
+package com.github.firmwehr.gentle.firm.construction;
+
+import com.github.firmwehr.gentle.semantic.ast.LocalVariableDeclaration;
+import com.github.firmwehr.gentle.semantic.ast.SField;
+import com.github.firmwehr.gentle.semantic.ast.SMethod;
+import com.github.firmwehr.gentle.semantic.ast.basictype.SIntType;
+import com.github.firmwehr.gentle.semantic.ast.type.SVoidType;
+import firm.ClassType;
+import firm.Entity;
+import firm.MethodType;
+import firm.Type;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * In firm most things identified by names (fields and methods in Gentle currently) are {@link Entity entities}. To use
+ * our own field and method declarations in the firm graph we need to translate them to firm entities first.
+ * <br>
+ * <p>This class converts fields and methods to singleton firm entities,</p>
+ */
+public class EntityHelper {
+	private final Map<SField, Entity> fieldEntities;
+	private final Map<SMethod, Entity> methodEntities;
+	private final TypeHelper typeHelper;
+
+	public EntityHelper(TypeHelper typeHelper) {
+		this.typeHelper = typeHelper;
+		this.fieldEntities = new HashMap<>();
+		this.methodEntities = new HashMap<>();
+	}
+
+
+	public void setFieldEntity(SField field) {
+		ClassType ownerType = typeHelper.getClassType(field.classDecl());
+		Type ownType = typeHelper.getType(field.type());
+		fieldEntities.put(field, new Entity(ownerType, field.name().ident(), ownType));
+	}
+
+	public Entity getEntity(SField field) {
+		return fieldEntities.get(field);
+	}
+
+	public Entity getEntity(StdLibEntity entity) {
+		return entity.getEntity();
+	}
+
+	public Entity computeMethodEntity(SMethod method) {
```

Comment:
Why `computeMethodEntity` but `setFieldEntity`?

Reply:
It returns an entity, the field one just creates one (returns void). I am open for suggestions for better names though

Reply:
Ah, I missed the return value